.github/workflows/release.yaml

```diff
@@ -74,3 +109,94 @@ jobs:
         uses: github/codeql-action/upload-sarif@v2
         with:
           sarif_file: snyk.sarif
+
+  binary-provenance:
+    needs: [ goreleaser ]
+    permissions:
+      actions: read # To read the workflow path.
+      id-token: write # To sign the provenance.
+      contents: write # To add assets to a release.
+
+    # Note: this _must_ be referenced by tag. See: https://github.com/slsa-framework/slsa-verifier/issues/12
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.9.0
+
+    with:
+      base64-subjects: "${{ needs.goreleaser.outputs.hashes }}"
+      upload-assets: true # upload to a new release
+
```

Comment:
Could we also add an `sbom-provenance` stage as well so that we have an intoto manifest  of the sbom artifacts?

```
sbom-provenance:
  needs: [generate-sbom]
  permissions:
    actions: read
    id-token: write
    contents: write
  # Must be refernced by a tag. https://github.com/slsa-framework/slsa-github-generator/blob/main/internal/builders/container/README.md#referencing-the-slsa-generator
  uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.9.0
  with:
    base64-subjects: "${{ hashes of sbom artifacts }}" # <- sbom hashes
    provenance-name: "openfga-sbom.intoto.jsonl"
    upload-assets: true
```

Developers should be able to verify the sbom artifacts, checksums,

Reply:
Not sure the best way to do that, but in general we want an intoto provenance of all of the artifacts, including the sboms. For example

```
openfga_1.3.5_darwin_arm64.tar.gz
openfga_1.3.5_darwin_arm64.sbom.tar.gz
openfga_1.3.5_darwin_arm64.intoto.jsonl
openfga_1.3.5_darwin_arm64-sbom.intoto.jsonl
```